### PR TITLE
Hotfix/change php reqs

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/HRIN-731

- Add github actions for PR
- Apply coding standards
- Add compatibility for php8.1

### CODE REVIEW

File: .github/workflows/push.yaml
Comment (rimi-itk) [inline .github/workflows/push.yaml:1]: We usually have only the `pr.yml` workflow file. We should probable remove `push.yaml`.

File: .github/workflows/pr.yml
Comment (rimi-itk) [inline .github/workflows/pr.yml:1]: Consider renaming to `pr.yaml` (as we usually name this file).

File: .php-cs-fixer.dist.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+$finder = PhpCsFixer\Finder::create()
+  ->in(__DIR__.'/{src,tests}')
+;
+
+$header =<<<'HEADER'
+This file is part of itk-dev/serviceplatformen.
+(c) 2020 ITK Development
+This source file is subject to the MIT license.
+HEADER;
```
Comment (rimi-itk) [inline .php-cs-fixer.dist.php]: We should not set the header in all files – and not to something containing “serviceplatformen”. The header does not really add any value. At least we should write `(c) 2020– ITK Development`.

File: composer.json [hunk 2]
```diff
@@ -3,15 +3,18 @@
     "description": "PHP client for pretix api",
     "type": "library",
     "require": {
-        "php": "^7.2",
+        "php": "^7.2||^8.1",
```
Comment (rimi-itk) [inline composer.json]: Remove `^7.2`.

File: phpcs.xml.dist [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <config name="drupal_core_version" value="9"/>
+
+    <rule ref="Drupal">
+        <!-- <exclude name="Drupal.Files.TxtFileLineLength.TooLong"/> -->
+        <!-- We want to be able to use "package" and "version" in our custom modules -->
+        <exclude name="Drupal.InfoFiles.AutoAddedKeys.Project"/>
+        <exclude name="Drupal.InfoFiles.AutoAddedKeys.Version"/>
+    </rule>
+
+    <rule ref="DrupalPractice"/>
```
Comment (rimi-itk) [inline phpcs.xml.dist]: This is not a Drupal module. We could skip using code sniffer and only use PHP Cs Fixer.

File: src/Client.php [hunk 4]
```diff
@@ -1,10 +1,8 @@
 <?php
 
-/*
- * This file is part of itk-dev/pretix-api-client-php.
- *
+/**
+ * This file is part of itk-dev/serviceplatformen.
```
Comment (rimi-itk) [inline src/Client.php]: Not really …

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: .github/workflows/push.yaml
Comment (martinyde) [reply .github/workflows/push.yaml:1]: Removed.

File: .github/workflows/pr.yml
Comment (martinyde) [reply .github/workflows/pr.yml:1]: Changed

File: .php-cs-fixer.dist.php (hunk 1 above)
Comment (martinyde) [reply .php-cs-fixer.dist.php]: Removed

File: composer.json (hunk 2 above)
Comment (martinyde) [reply composer.json]: Removed.

File: phpcs.xml.dist (hunk 3 above)
Comment (martinyde) [reply phpcs.xml.dist]: Removed

File: src/Client.php (hunk 4 above)
Comment (martinyde) [reply src/Client.php]: Removed.

File: phpstan.neon [hunk 5]
```diff
@@ -0,0 +1,9 @@
+parameters:
+  paths:
+    - src
+    - tests
+  level: 0
+  customRulesetUsed: true
+  ignoreErrors:
+      - '#\Drupal calls should be avoided in classes, use dependency injection instead#'
```
Comment (rimi-itk) [inline phpstan.neon]: I think we should fix this issue if possible rather than ignore the complaint.

File: src/Client.php [hunk 6]
```diff
@@ -513,7 +507,7 @@ public function getWebhooks(): EntityCollectionInterface
      *
      * @return \ItkDev\Pretix\Api\Response
      */
-    public function getWebhook($id): Response
+    public function getWebhook($id): HttpResponse
```
Comment (rimi-itk) [inline src/Client.php]: This return type does not match the type declared by `@return`. One of them must be wrong.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**